### PR TITLE
team: new action icons and responsive icon-only mode (fixes #7169)

### LINKED ISSUE #7169: Team/Enterprises: Make the Action buttons to display icons only for mobile screens
Suggestion to make the action buttons in the enterprises & teams display page to be icons only to avoid text being cutoff for smaller screens

image

### PR DESCRIPTION
**Pull request by Mutugiii**
- add the checkMobile service to the teams
- make teams action buttons to display icons in mobile screens 
- separate the styles from the component file to a separate style sheet(Past 12 lines as in style-guide.md)

Fixes #7169

### CODE REVIEW

**Comment by Mutugiii** [comment]
On hold for #7166

File: src/app/teams/teams.component.html [hunk 1]
```diff
@@ -64,21 +64,63 @@ <h3>
...
+                <ng-template #mobileView>
+                  <mat-icon>hourglass_empty</mat-icon>
+                </ng-template>
+              </button>
             </ng-container>
-            <button *ngSwitchCase="'unrelated'" mat-raised-button color="primary" class="margin-lr-3" (click)="requestToJoin(element.doc); $event.stopPropagation()">
-              <ng-container i18n>Request to Join</ng-container>
+            <button mat-raised-button color="primary" planetFeedback [feedbackOf]="{'state': mode + 's', 'item': element.doc._id}" (click)="$event.stopPropagation()" i18n>
+              <ng-container *ngIf="!isMobile; else mobileView">
+                <mat-icon>feedback</mat-icon> Feedback
```
Comment (lmmrssa) [inline src/app/teams/teams.component.html:117]: only label can be part of if condition

File: src/app/teams/teams.component.html [hunk 2]
```diff
@@ -64,21 +64,63 @@ <h3>
...
+                  </ng-container>
+                  <ng-template #mobileView>
+                    <mat-icon>edit</mat-icon>
+                  </ng-template>
+                </button>
               </ng-container>
-              <button *ngIf="element.isLeader" mat-raised-button color="primary" (click)="addTeam(element.doc); $event.stopPropagation()" i18n><mat-icon>edit</mat-icon> Edit</button>
+              <button *ngSwitchCase="'unrelated'" mat-raised-button color="primary" class="margin-lr-3" (click)="requestToJoin(element.doc); $event.stopPropagation()">
+                <ng-container *ngIf="!isMobile; else mobileView">
+                  <mat-icon>add_circle</mat-icon> Request to Join
```
Comment (lmmrssa) [inline src/app/teams/teams.component.html:100]: label only in condition

File: src/app/teams/teams.component.html [hunk 3]
```diff
@@ -64,21 +64,63 @@ <h3>
...
+                </ng-container>
+                <ng-template #mobileView>
+                  <mat-icon>add_circle</mat-icon>
+                </ng-template>
+              </button>
+              <button *ngSwitchCase="'requesting'" mat-raised-button color="primary" disabled i18n>
+                <ng-container *ngIf="!isMobile; else mobileView">
+                  <mat-icon>hourglass_empty</mat-icon> Request pending
```
Comment (lmmrssa) [inline src/app/teams/teams.component.html:108]: label only in condition

File: src/app/teams/teams.component.html [hunk 4]
```diff
@@ -64,21 +64,63 @@ <h3>
...
-          <button *planetAuthorizedRoles="''" mat-raised-button color="primary" class="margin-lr-3" (click)="archiveClick(element.doc); $event.stopPropagation()" i18n><mat-icon>delete_forever</mat-icon> Delete</button>
-          <ng-container [ngSwitch]="element.userStatus" *ngIf="user.isUserAdmin || user.roles.length">
-            <ng-container *ngSwitchCase="'member'">
-              <ng-container *ngIf="!element.isLeader">
-                <button mat-raised-button color="primary" class="margin-lr-3" (click)="openLeaveDialog(element.doc, element.membershipDoc); $event.stopPropagation()" i18n><mat-icon>remove_circle</mat-icon> Leave</button>
+        <mat-cell *matCellDef="let element" class>
+          <div [ngClass]="{'horizontal-align': isMobile, 'button-container': isMobile}">
+            <button *planetAuthorizedRoles="''" mat-raised-button color="primary" class="margin-lr-3" (click)="archiveClick(element.doc); $event.stopPropagation()" i18n>
+              <ng-container *ngIf="!isMobile;else mobileView">
+                <mat-icon>delete_forever</mat-icon> Delete
```
Comment (lmmrssa) [inline src/app/teams/teams.component.html:71]: label only in condition

**Review by lmmrssa — CHANGES_REQUESTED** [review]

**Review by lmmrssa — APPROVED** [review]

**Pull request merged**